### ISSUE TITLE
Flow's untyped input defaults can lead to runtime errors

### ISSUE THREAD

**Issue by yuri1969** [kestra-io/kestra#5590]
### Describe the issue

*TL;DR flow's input defaults should be correctly typed.*

---

Currently the flow's input `defaults` field is not typed - it is a simple `Object`. This can lead to unexpected runtime errors in situations when a wrongly typed default is effectively applied and a `validator` is defined.

For example, a validated `STRING` input with `defaults` being an integer fails since the validator can not be applied on an integer:

```yaml
id: badDefaults
namespace: io.kestra.tests

inputs:
  - id: intDefaultsInStrInput
    type: STRING
    validator: '\d+'
    defaults: 100
    required: false

tasks:
  - id: date
    type: io.kestra.plugin.core.debug.Return
    format: "{{taskrun.startDate}}"
```

Execute the flow above using the API and omit the flow input completely - the execution fails on validation.

---

The same behavior can be demonstrated by the following unit test: [mismatchedDefaults.patch](https://github.com/user-attachments/files/17471405/mismatchedDefaults.patch)

---

It feels like this issue should be easily solved by switching the `Input`s `defaults` attribute type from `Object` to the generic type `T` - [Input.java L81](https://github.com/kestra-io/kestra/blob/dd48ad869cc3e6a5da93f3a18aff8b2d1e3313d8/core/src/main/java/io/kestra/core/models/flows/Input.java#L81). It enables JSON Schema generation for the `defaults`.

However, it breaks at least the `TIME` input a bit - JSON Schema's `time` requires timezone. Also the value itself is not validated.

### Environment

- Kestra Version: v0.20-SNAPSHOT
- Operating System (OS/Docker/Kubernetes):
- Java Version (if you don't run kestra in Docker): 21 Temurin

**Comment by Malaydewangan09** [comment]
Hey @loicmathieu 
Can I pick this up?
Thanks!

**Comment by loicmathieu** [comment]
@Malaydewangan09 yes, I assigned you to the issue.

**Comment by Malaydewangan09** [comment]
@loicmathieu Thanks!
